File: backend/apps/academics/models/topic.py
from django.db import models


class Topic(models.Model):
    """Maps to `topic` in db_current."""

    id = models.AutoField(primary_key=True)
    session_id = models.IntegerField(blank=False, null=False, db_index=True)
    lesson_id = models.IntegerField(blank=False, null=False, db_index=True)
    name = models.CharField(max_length=255)
    status = models.IntegerField()
    complete_date = models.DateField(blank=True, null=True)
    created_at = models.DateTimeField()

    class Meta:
        managed = False
        db_table = "topic"

    def __str__(self):
        return f"Topic {self.pk}"

File: frontend/src/types/academics/topic.ts
export interface Topic {
  id: number;
  session_id: number;
  lesson_id: number;
  name: string;
  status: number;
  complete_date: string | null;
  created_at: string;
  subject_name?: string | null;
  subject_group_name?: string | null;
  class_section_name?: string | null;
  lesson_name?: string | null;
  subject_group_id?: number | null;
  subject_id?: number | null;
  class_section_id?: number | null;
}

export interface TopicCreatePayload {
  session_id: number;
  lesson_id: number;
  name: string;
  status?: number;
  complete_date?: string | null;
}

export interface TopicUpdatePayload {
  session_id?: number;
  lesson_id?: number;
  name?: string;
  status?: number;
  complete_date?: string | null;
}
